test/suite/test_layered48.py

```diff
@@ -0,0 +1,127 @@
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, os.path, shutil, time, wiredtiger, wttest
+from helper_disagg import DisaggConfigMixin, disagg_test_class, gen_disagg_storages
+from wtscenario import make_scenarios
+from wiredtiger import stat
+
+# test_layered48.py
+#    Ensure overflow keys and values are not being generated in disaggregated storage.
+
+@disagg_test_class
+class test_layered48(wttest.WiredTigerTestCase, DisaggConfigMixin):
+    nitems = 500
+    key_to_update = 0
+    num_updates = 10
+
+    conn_base_config = 'statistics=(all),' \
+                     + 'statistics_log=(wait=1,json=true,on_close=true),' \
+                     + 'precise_checkpoint=true,disaggregated=(page_log=palm),'
+    conn_config = conn_base_config + 'disaggregated=(role="leader")'
+
+    create_session_config = 'key_format=S,value_format=S'
+
+    table_name = "test_layered48"
+
+    disagg_storages = gen_disagg_storages('test_layered48', disagg_only = True)
+    scenarios = make_scenarios(disagg_storages, [
+        ('layered', dict(prefix='layered:')),
+        ('shared', dict(prefix='table:')),
+    ])
+
+    # Load the page log extension, which has object storage support
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        DisaggConfigMixin.conn_extensions(self, extlist)
+
+    # Custom test case setup
+    def early_setup(self):
+        os.mkdir('follower')
+        # Create the home directory for the PALM k/v store, and share it with the follower.
+        os.mkdir('kv_home')
+        os.symlink('../kv_home', 'follower/kv_home', target_is_directory=True)
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    # Test overflow keys and values.
+    def test_layered18(self):
+
+        # Create table
+        self.uri = self.prefix + self.table_name
+        table_config = self.create_session_config
+        if not self.uri.startswith('layered'):
+            table_config += ',block_manager=disagg,log=(enabled=false)'
+        self.session.create(self.uri, table_config)
+
+        # Put big data to the table
+        key_prefix1 = 'aabbcc' * 1000
```

Comment:
I think we should avoid using this kind of values. The wt compression will be very effective in this case. Better to use a random value.